ui/components/ErrorBoundary.js

```diff
@@ -1,40 +1,43 @@
-import { Button } from '@material-ui/core';
-import React from 'react';
-
-/**
- * ErrorBoundary is a React component that catches JavaScript errors in its child components and renders a fallback UI when an error occurs.
- * It should be used as a wrapper around components that might throw errors.
- */
-class ErrorBoundary extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = { hasError: false, error: null };
-  }
+import Button from '@mui/material/Button';
+import React, { useState, useEffect } from 'react';
 
-  /** Update state so the next render will show the fallback UI. */
-  static getDerivedStateFromError(error) {
-    return { hasError: true, error: error };
-  }
+const ErrorBoundary = ({ children }) => {
+  const [hasError, setHasError] = useState(false);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    const handleErrors = (error) => {
+      setHasError(true);
+      setError(error);
+    };
 
-  resetErrorBoundary = () => {
-    this.setState({ hasError: false, error: null });
+    const resetErrorBoundary = () => {
+      setHasError(false);
+      setError(null);
+    };
+
+    window.addEventListener('error', handleErrors);
+    return () => window.removeEventListener('error', handleErrors);
```

Comment:
We don't need to listen to event, @devhimanshuu is there specific reason to use this, how you make sure we don't use error boundaries for all error which can be ignored

Reply:
I agree what you are saying but, i add these event listener for some global errors or unexpected error.  may be there is some optimized way of handling these errors.